### PR TITLE
T-19089 feat: updated listener types

### PR DESCRIPTION
**Pull request by commoddity**

### CODE REVIEW

File: types/common.go [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+	ActionInsert Action = "INSERT"
+	ActionUpdate Action = "UPDATE"
+	ActionDelete Action = "DELETE"
+)
+
+type SavedOnDB interface {
+	Table() Table
+}
+
+func (t *PortalApp) Table() Table {
```
Comment (adshmh) [inline types/common.go:53]: Could we keep the method(s) for each type in the same place as the declaration of the type?

File: types/common.go [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+	return TablePortalApps
+}
+
+func (t *AppAAT) Table() Table {
```
Comment (adshmh) [inline types/common.go:57]: Same comment re methods and struct declaration location in the code.

**Review by adshmh — CHANGES_REQUESTED** [review]

File: types/common.go (hunk 1 above)
Comment (commoddity) [reply types/common.go:53]: Would prefer not to personally so as not to clutter the individual fields when these methods seem more logically connected by their function to the listener. WDYT @pablocampogo ?

Comment (commoddity) [reply types/common.go:52]: OK @pablocampogo agrees with you ;) I'll move them to the same file as the struct declarations

**Review by adshmh — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**